cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -283,6 +284,16 @@ public ServiceTicket grantServiceTicket(
         // This throws if no suitable policy is found
         getAuthenticationSatisfiedByPolicy(ticketGrantingTicket, new ServiceContext(service, registeredService));
 
+        final List<Authentication> authentications = ticketGrantingTicket.getChainedAuthentications();
+        final Principal principal = authentications.get(authentications.size() - 1).getPrincipal();
+
+        final Map<String, Object> principalAttrs = registeredService.getAttributeReleasePolicy().getAttributes(principal);
+        if (!registeredService.getAuthorizationStrategy().isServiceAccessAuthorizedForPrincipal(principalAttrs)) {

```

Comment:
This is done on purpose, because principal attributes may not be immediately the same. If the attributes are cached, you want to retrieve the latest version. If they are not cached, you will receive the same map. Because we can't call principal.getAttributes() to always retrieve the current/latest set, we pass in a map instead that is pre-processed.